pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -610,30 +674,37 @@ func (gc *GarbageCollector) attemptToDeleteItem(ctx context.Context, item *node)
 			// otherwise, default to background.
 			policy = metav1.DeletePropagationBackground
 		}
-		klog.V(2).InfoS("Deleting object", "object", klog.KRef(item.identity.Namespace, item.identity.Name),
-			"objectUID", item.identity.UID, "kind", item.identity.Kind, "propagationPolicy", policy)
+		logger.V(2).Info("Deleting object",
+			"object", klog.KRef(item.identity.Namespace, item.identity.Name),
+			"objectUID", item.identity.UID,
+			"kind", item.identity.Kind,
+			"propagationPolicy", policy,
+		)
 		return gc.deleteObject(item.identity, &policy)
 	}
 }
 
 // process item that's waiting for its dependents to be deleted
-func (gc *GarbageCollector) processDeletingDependentsItem(item *node) error {
+func (gc *GarbageCollector) processDeletingDependentsItem(logger klog.Logger, item *node) error {
 	blockingDependents := item.blockingDependents()
 	if len(blockingDependents) == 0 {
-		klog.V(2).Infof("remove DeleteDependents finalizer for item %s", item.identity)
-		return gc.removeFinalizer(item, metav1.FinalizerDeleteDependents)
+		logger.V(2).Info("remove DeleteDependents finalizer for item", "item", item.identity)
+		return gc.removeFinalizer(logger, item, metav1.FinalizerDeleteDependents)
 	}
 	for _, dep := range blockingDependents {
 		if !dep.isDeletingDependents() {
-			klog.V(2).Infof("adding %s to attemptToDelete, because its owner %s is deletingDependents", dep.identity, item.identity)
+			logger.V(2).Info("adding dependent to attemptToDelete, because its owner is deletingDependents",
+				"item", item.identity,
+				"dependent", dep.identity,
```

Comment:
Also here: I don't know how large `dep` is, but perhaps it should get logged?

Reply:
OTOH, that makes the log larger than it used to be, something that goes beyond merely translating the calls.

Reply:
Keeping dep.identity